### ISSUE TITLE
tf2onnx==1.9.3 installation error in python 3.10.0

### ISSUE THREAD

**Issue by hanzigs** [onnx/tensorflow-onnx#1900]
Hi,
I am upgrading my python to 3.10.0, currently using 3.9.5 where tf2onnx==1.9.2 working good.
I installed 1.9.2 and 1.9.3 in py3.10.0, I am getting following error while installing
May I have some help please
```
Collecting tf2onnx==1.9.3
  Downloading tf2onnx-1.9.3-py3-none-any.whl (435 kB)
     |████████████████████████████████| 435 kB 6.8 MB/s
Collecting flatbuffers~=1.12
  Using cached flatbuffers-1.12-py2.py3-none-any.whl (15 kB)
Requirement already satisfied: requests in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from tf2onnx==1.9.3) (2.27.1)
Requirement already satisfied: six in c:\users\pg\anaconda3\envs\automlpy3100t2\lib\site-packages (from tf2onnx==1.9.3) (1.16.0)
Requirement already satisfied: numpy>=1.14.1 in c:\users\pg\anaconda3\envs\py3100t2\lib\site-packages (from tf2onnx==1.9.3) (1.21.5)
Collecting onnx>=1.4.1
  Downloading onnx-1.11.0.tar.gz (9.9 MB)
     |████████████████████████████████| 9.9 MB ...
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_8901be70db104e67b2b69214bcf2f4d3\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_8901be70db104e67b2b69214bcf2f4d3\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\pg\AppData\Local\Temp\pip-pip-egg-info-duzv_rxn'
         cwd: C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_8901be70db104e67b2b69214bcf2f4d3\
    Complete output (6 lines):
    fatal: not a git repository (or any of the parent directories): .git
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_8901be70db104e67b2b69214bcf2f4d3\setup.py", line 86, in <module>
        assert CMAKE, 'Could not find "cmake" executable!'
    AssertionError: Could not find "cmake" executable!
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/fd/b7/fccddff4d1873074605ff08acc812202b4a849cf4925b1f6ed5eeba575c4/onnx-1.11.0.tar.gz#sha256=eca224c7c2c8ee4072a0743e4898a84a9bdf8297b5e5910a2632e4c4182ffb2a (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading onnx-1.10.2.tar.gz (9.9 MB)
     |████████████████████████████████| 9.9 MB 6.8 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\pg~1\AppData\Local\Temp\tmpzc6_vno7'
       cwd: C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_db1a2ad44aec4dafaec0fceeb5be2b3a
  Complete output (19 lines):
  fatal: not a git repository (or any of the parent directories): .git
  Traceback (most recent call last):
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 349, in <module>
      main()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-anyb2oz8\overlay\Lib\site-packages\setuptools\build_meta.py", line 177, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-anyb2oz8\overlay\Lib\site-packages\setuptools\build_meta.py", line 159, in _get_build_requires
      self.run_setup()
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-anyb2oz8\overlay\Lib\site-packages\setuptools\build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-anyb2oz8\overlay\Lib\site-packages\setuptools\build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 86, in <module>
      assert CMAKE, 'Could not find "cmake" executable!'
  AssertionError: Could not find "cmake" executable!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/16/6a/bdae938babb4bc23de7b599439f3d1f1179748385e4ced099f3b4cb646bd/onnx-1.10.2.tar.gz#sha256=24d73ca7dfd7e6c7339944f89554b4010719899337924fca1447d8f1b5db50d6 (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\autoMLpy3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\pg~1\AppData\Local\Temp\tmpzc6_vno7' Check the logs for full command output.
  Downloading onnx-1.10.1.tar.gz (10.0 MB)
     |████████████████████████████████| 10.0 MB ...
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_c4c0d1887a6a499bb902ffb07728cde8\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_c4c0d1887a6a499bb902ffb07728cde8\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\pg\AppData\Local\Temp\pip-pip-egg-info-brg2sjsv'
         cwd: C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_c4c0d1887a6a499bb902ffb07728cde8\
    Complete output (6 lines):
    fatal: not a git repository (or any of the parent directories): .git
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_c4c0d1887a6a499bb902ffb07728cde8\setup.py", line 86, in <module>
        assert CMAKE, 'Could not find "cmake" executable!'
    AssertionError: Could not find "cmake" executable!
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/18/ec/d1d74c6355a139a633d1335620e6d891e3a3556889b3d0ede878ab53fc11/onnx-1.10.1.tar.gz#sha256=9d941ba76cab55db8913ecad9dc50cefeb368460f6338a91783a5d7643f3a044 (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading onnx-1.10.0.tar.gz (10.0 MB)
     |████████████████████████████████| 10.0 MB ...
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_35706237d6d249978f3c7629dcbfe1fa\\setup.py'"'"'; __file__='"'"'C:\\Users\\pg\\AppData\\Local\\Temp\\pip-install-8tt6xzoy\\onnx_35706237d6d249978f3c7629dcbfe1fa\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\pg\AppData\Local\Temp\pip-pip-egg-info-q__pwdtx'
         cwd: C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_35706237d6d249978f3c7629dcbfe1fa\
    Complete output (6 lines):
    fatal: not a git repository (or any of the parent directories): .git
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_35706237d6d249978f3c7629dcbfe1fa\setup.py", line 86, in <module>
        assert CMAKE, 'Could not find "cmake" executable!'
    AssertionError: Could not find "cmake" executable!
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/cf/98/82671f253342463e63fa972713f2bab1d936dd178455501d36ff288b209f/onnx-1.10.0.tar.gz#sha256=5919dcb0987265f8e9da4776cce7431f6d1db03beb561419739aa0874514d2d1 (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading onnx-1.9.0.tar.gz (9.8 MB)
     |████████████████████████████████| 9.8 MB ...
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\autoMLpy3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\pg~1\AppData\Local\Temp\tmpzs82i220'
       cwd: C:\Users\PremGeorgeAlphonse\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_a38f2af111ef4ce797146684cc959ef1
  Complete output (19 lines):
  fatal: not a git repository (or any of the parent directories): .git
  Traceback (most recent call last):
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 349, in <module>
      main()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-87vfiacr\overlay\Lib\site-packages\setuptools\build_meta.py", line 177, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-87vfiacr\overlay\Lib\site-packages\setuptools\build_meta.py", line 159, in _get_build_requires
      self.run_setup()
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-87vfiacr\overlay\Lib\site-packages\setuptools\build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-87vfiacr\overlay\Lib\site-packages\setuptools\build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 86, in <module>
      assert CMAKE, 'Could not find "cmake" executable!'
  AssertionError: Could not find "cmake" executable!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/73/e9/5b953497c0e36df589fc60cc6c6b35a65eb67d9ad1e45a9163663e43426e/onnx-1.9.0.tar.gz#sha256=a9055a654c4dd87b45ce59f1e08371686937cad554471eadf6826df1bb125b7a (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: 'C:\Users\pg\anaconda3\envs\autoMLpy3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\PREMGE~1\AppData\Local\Temp\tmpzs82i220' Check the logs for full command output.
  Downloading onnx-1.8.1.tar.gz (5.2 MB)
     |████████████████████████████████| 5.2 MB ...
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\autoMLpy3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\pg~1\AppData\Local\Temp\tmp17tsm9ik'
       cwd: C:\Users\pg\AppData\Local\Temp\pip-install-8tt6xzoy\onnx_27dd092724a54959824cd2a1406e05e9
  Complete output (19 lines):
  fatal: not a git repository (or any of the parent directories): .git
  Traceback (most recent call last):
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 349, in <module>
      main()
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-zc7gfl91\overlay\Lib\site-packages\setuptools\build_meta.py", line 177, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-zc7gfl91\overlay\Lib\site-packages\setuptools\build_meta.py", line 159, in _get_build_requires
      self.run_setup()
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-zc7gfl91\overlay\Lib\site-packages\setuptools\build_meta.py", line 281, in run_setup
      super(_BuildMetaLegacyBackend,
    File "C:\Users\pg\AppData\Local\Temp\pip-build-env-zc7gfl91\overlay\Lib\site-packages\setuptools\build_meta.py", line 174, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 75, in <module>
      assert CMAKE, 'Could not find "cmake" executable!'
  AssertionError: Could not find "cmake" executable!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/2b/8d/c924f54e19ec71543c18658d86e8337579d5a784bd524b7f18e2e2cd6f17/onnx-1.8.1.tar.gz#sha256=9d65c52009a90499f8c25fdfe5acda3ac88efe0788eb1d5f2575a989277145fb (from https://pypi.org/simple/onnx/). Command errored out with exit status 1: 'C:\Users\pg\anaconda3\envs\py3100t2\python.exe' 'C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' get_requires_for_build_wheel 'C:\Users\pg~1\AppData\Local\Temp\tmp17tsm9ik' Check the logs for full command output.
  Downloading onnx-1.8.0.tar.gz (5.2 MB)
     |████████████████████████████████| 5.2 MB 6.4 MB/s
  Installing build dependencies ... canceled
ERROR: Operation cancelled by user
Traceback (most recent call last):
  File "C:\Users\pg\anaconda3\envs\py3100t2\Scripts\pip-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\cli\main.py", line 70, in main
    return command.main(cmd_args)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\cli\base_command.py", line 97, in main
    with self.main_context():
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 142, in __exit__
    next(self.gen)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\cli\command_context.py", line 19, in main_context
    with self._main_context:
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 576, in __exit__
    raise exc_details[1]
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\utils\temp_dir.py", line 75, in tempdir_registry
    yield _tempdir_registry
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 561, in __exit__
    if cb(*exc_details):
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 142, in __exit__
    next(self.gen)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\utils\temp_dir.py", line 32, in global_tempdir_manager
    with ExitStack() as stack:
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 576, in __exit__
    raise exc_details[1]
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\contextlib.py", line 561, in __exit__
    if cb(*exc_details):
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\utils\temp_dir.py", line 165, in __exit__
    self.cleanup()
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\utils\temp_dir.py", line 184, in cleanup
    rmtree(self._path)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 326, in wrapped_f
    return self(f, *args, **kw)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 406, in __call__
    do = self.iter(retry_state=retry_state)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 351, in iter
    return fut.result()
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\concurrent\futures\_base.py", line 438, in result
    return self.__get_result()
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\concurrent\futures\_base.py", line 390, in __get_result
    raise self._exception
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 409, in __call__
    result = fn(*args, **kwargs)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\site-packages\pip\_internal\utils\misc.py", line 135, in rmtree
    shutil.rmtree(dir, ignore_errors=ignore_errors, onerror=rmtree_errorhandler)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\shutil.py", line 739, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\shutil.py", line 612, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\shutil.py", line 612, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\shutil.py", line 612, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  [Previous line repeated 2 more times]
  File "C:\Users\pg\anaconda3\envs\py3100t2\lib\shutil.py", line 615, in _rmtree_unsafe
    os.unlink(fullname)
```

**Comment by hwangdeyu** [comment]
Thanks! Will support python 3.10 later.

**Comment by hwangdeyu** [comment]
Waitting for [ONNX PYPI](https://pypi.org/project/onnx/#files) Python 3.10 support.
https://github.com/onnx/onnx/issues/4110

**Comment by jcwchen** [comment]
FYI: ONNX has released ONNX yesterday and Python 3.10 has been supported. Thanks!

**Comment by hwangdeyu** [comment]
@hanzigs You can install tf2onnx successfully in Python 3.10 now, but ONNXRUNTIME would support 3.10 in next release.
